cloud_governance/policy/policy_operations/aws/tag_non_cluster/tag_non_cluster_resources.py

```diff
@@ -141,24 +147,33 @@ def update_snapshots(self, snapshots: list = None):
             snapshot_id = snapshot.get('SnapshotId')
             tags = snapshot.get('Tags')
             if not self.validate_existing_tag(tags=tags):
-                username = self.get_username(start_time=snapshot.get('StartTime'), resource_id=snapshot_id, resource_type='AWS::EC2::Snapshot', tags=tags)
+                username = self.get_username(start_time=snapshot.get('StartTime'), resource_id=snapshot_id,
+                                             resource_type='AWS::EC2::Snapshot', tags=tags)
                 if 'vm_import_image' in username:
                     start_time = snapshot.get('StartTime') + timedelta(seconds=5)
                     end_time = start_time + timedelta(minutes=30)
-                    assume_username = self.get_username(start_time=start_time, resource_id=snapshot_id, resource_type='AWS::EC2::Snapshot', tags=tags, end_time=end_time)
+                    assume_username = self.get_username(start_time=start_time, resource_id=snapshot_id,
+                                                        resource_type='AWS::EC2::Snapshot', tags=tags,
+                                                        end_time=end_time)
                     if assume_username:
                         username = assume_username
                 search_tags = []
                 if not username:
                     if snapshot.get('Description') and 'Created' in snapshot.get('Description'):
-                        image_tags, username = self._get_tags_from_snapshot_description_images(description=snapshot.get('Description'))
+                        image_tags, username = self._get_tags_from_snapshot_description_images(
+                            description=snapshot.get('Description'))
                         if not username:
-                            instance_id = snapshot.get('Description').split(" ")[2].split("(")[1][:-1]
-                            instances = self._get_instances_data(instance_id)
-                            if instances:
-                                for item in instances:
-                                    if item.get('InstanceId') == instance_id:
-                                        item_tags, username = self._get_tags_from_instance_item(instance_item=item)
+                            if "i-" in snapshot.get('Description'):
+                                try:
```

Comment:
Are you ok with fetching it using a hardcoded string, or is there a safer option to filter it?

Reply:
yes, we need to fetch it from snapshot